### PR TITLE
<Icons/> - use new set of icons:

### PR DESCRIPTION
**Pull request by Faradey27**
Plan of commits:
 - [x] created script to proxy wix-ui-icons-common/general icons to wix-style-react/icons folder
 - [x] update package.json with new build script and remove old icons
 - [x] verify that the recently added icons in WSR are also in wix-ui-icons-common/general
 - [x] update icons storybook example
 - [x] update and fix all storybooks examples that depend on Icons
 - [x] add a icons section in the contribution doc (explain the automated script)
 - [x] provide a migration doc. both for consumers and WSR devs
 - [x] * create a migration codemod
### What changed

- Migrated to new icons and new working flow with icons

### Why it changed

- because of wsr-issue
- to improve users and dev experience 

---

- [x] Change is tested
- [x] Change is documented
- [x] Build is green
- [x] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,11 +70,13 @@
     "eslint-plugin-react": "^6.8.0",
     "express": "^4.15.2",
     "eyes.it": "^2.0.0",
+    "fs-extra": "^6.0.1",
```
Comment (shlomitc) [inline package.json]: can we remove it and just use node's `fs`? as this probably needs to be a dependency of the project and not just dev-dep

File: package.json [hunk 2]
```diff
@@ -68,11 +70,13 @@
...
     "gh-pages-auto-release": "^1.1.2",
     "github-markdown-css": "^2.4.1",
     "highlight.js": "^9.8.0",
     "identity-obj-proxy": "^3.0.0",
     "import-path": "^1.0.0",
+    "jscodeshift": "^0.5.0",
```
Comment (shlomitc) [inline package.json]: should be a dependency and not dev-dep

File: scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js [hunk 3]
```diff
@@ -0,0 +1,360 @@
+import React, {Component} from 'react';
```
Comment (shlomitc) [inline scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js]: full example not needed, only the import part

File: src/AutoCompleteComposite/AutoCompleteComposite.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import Label from '../Label';
 import AutoComplete from '../AutoComplete';
 import InputAreaWithLabelComposite from '../Composite/InputAreaWithLabelComposite/InputAreaWithLabelComposite';
 
-const AutoCompleteComposite = ({...props, children}) => (
+const AutoCompleteComposite = ({children, ...props}) => (
```
Comment (shlomitc) [inline src/AutoCompleteComposite/AutoCompleteComposite.js]: good catch, but it should be in a different pr

File: package.json (hunk 1 above)
Comment (Faradey27) [reply package.json]: in general we can, but i will need to create ugly recursive function to read all files from folder

File: scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js (hunk 3 above)
Comment (Faradey27) [reply scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js]: i think imports not enought, because we also need to change variable usage

in code i am doing search via AST for <OldIcon /> and rename it to <NewIcon /> and <Icons.oldIcon>  to <Icons.newIcon>

so it maybe usefull to have all file here

Comment (Faradey27) [reply scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js]: @shlomitc

File: src/AutoCompleteComposite/AutoCompleteComposite.js (hunk 4 above)
Comment (Faradey27) [reply src/AutoCompleteComposite/AutoCompleteComposite.js]: agree, the problem was, that AST build failed because of such order))

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -63,3 +63,16 @@ open localhost:6006
...
+* go to [wix-ui](https://github.com/wix/wix-ui) repo to [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) package
+* decide together with UX guild is it `system` or `common` icon
+* add your icon to correct package folder inside [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common)
+* create pull request for [wix-ui](https://github.com/wix/wix-ui) with your new icon
+* after merging your pull request inside [wix-ui](https://github.com/wix/wix-ui), your new icon will automatically appear inside [wix-style-react](https://github.com/wix/wix-style-react) with next wix-style-react release(wix-style-react has automation script, which take all icons from [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) and proxy them inside [wix-style-react](https://github.com/wix/wix-style-react))
+* your icon will be available in wix-style-react with such syntax `import MyAwesomeIcon from 'wix-style-react/icons/MyAwesomeIcon'` or `import MyAwesomeIcon from 'wix-style-react/icons/system/MyAwesomeIcon'`
```
Comment (shlomitc) [inline docs/CONTRIBUTING.md]: we actually shouldn't expose the system icons to our consumers.
@milkyfruit - is there any reason for a user to consume a system icon?

File: docs/NEW_ICONS_MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+* current icons do not have standart sizes
+* a lot of current icons are broken
+* it is tricky to import and to use current icons
+
+# How to use new icons?
+
+Example 1
+```
+  import Add from 'wix-style-react/icons/Add';
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: shouldn't it be `import Add from 'wix-style-react/new-icons/Add'`

File: docs/NEW_ICONS_MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  export default () => (
+    <div>
+      <Add />
+    </div>
+  )
+```
+
+Example 2
+```
+  import {Add} from 'wix-style-react/icons';
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: this should come with a warning that this will currently import all icons since tree shaking will be introduced only when WSR will be using yoshi2 (WIP)

File: docs/NEW_ICONS_MIGRATION.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  export default () => (
+    <div>
+      <Add size="3em"/>
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: I guess this should come with an explanation to the size/width/height props

File: docs/NEW_ICONS_MIGRATION.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+# How to migrate to new icons
+
+First of all, some names are changed and some icons are deprecated, so please visit http://electric-process.surge.sh/ and find new icon name, if your icon is deprecated - ask your UX/UI person to provide alternative from available icons.
+
+Now you have two choices:
+1) Do migration by hands based on mapping
+2) run our migration script
+```
+  migrate-to-icons-v2 --path=src/
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: `npx migrate-to-icons-v2 --path=...`
I prefer having it work without parameters so it will be easy to use.
you can mention it can work on a specific path if needed

File: docs/NEW_ICONS_MIGRATION.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  import {Add} from 'wix-style-react/icons';
+
+  export default () => (
+    <div>
+      <Add size="3em"/>
+    </div>
+  )
+```
+
+# How to migrate to new icons
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: should be higher in order IMHO

File: scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js (hunk 3 above)
Comment (shlomitc) [reply scripts/migrate-to-icons-v2/__testfixtures__/generalTransform.input.js]: ok

File: src/AutoCompleteComposite/AutoCompleteComposite.js (hunk 4 above)
Comment (shlomitc) [reply src/AutoCompleteComposite/AutoCompleteComposite.js]: what do you mean? which AST?

File: src/new-icons/README.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+> SVG Icon base type. The list of existing icons can be found [here]('https://wix-wix-style-react.surge.sh/?selectedKind=6.%20Common&selectedStory=6.5%20Icons&full=0&down=0&left=1&panelRight=0')
+
+## Properties
+
+| propName | propType | defaultValue | isRequired | description |
+|----------|----------|--------------|------------|-------------|
+| size | string | 1em | - | Set the size of the icon |
+| ***All other Props are passed to the SVG element*** | | | | |
+
+## Adding a new Icon
```
Comment (shlomitc) [inline src/new-icons/README.md]: this shouldn't be part of the usage, you can refer to the contribution section

File: src/AutoCompleteComposite/AutoCompleteComposite.js (hunk 4 above)
Comment (Faradey27) [reply src/AutoCompleteComposite/AutoCompleteComposite.js]: http://prntscr.com/jr3z3t

File: docs/NEW_ICONS_MIGRATION.md (hunk 9 above)
Comment (Faradey27) [reply docs/NEW_ICONS_MIGRATION.md]: ok, i will mention that it can accept parameters but by default it will work

File: docs/CONTRIBUTING.md [hunk 12]
```diff
@@ -63,3 +63,13 @@ open localhost:6006
...
+
+## Adding new icon
+
+* receive the approval for the new icon from one of the library's UXs [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)
+* go to [wix-ui](https://github.com/wix/wix-ui) repo to [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) package
+* add it to the system icons folder if it's an icon used internally inside a wix-style-react component, or to the general in case it is an icon to be used inside an app
+* add your icon to correct package folder inside [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common)
+* create pull request for [wix-ui](https://github.com/wix/wix-ui) with your new icon
+* after merging your pull request inside [wix-ui](https://github.com/wix/wix-ui), your new icon will automatically appear inside [wix-style-react](https://github.com/wix/wix-style-react) with next wix-style-react release(wix-style-react has automation script, which take all icons from [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) and proxy them inside [wix-style-react](https://github.com/wix/wix-style-react))
```
Comment (shlomitc) [inline docs/CONTRIBUTING.md]: it will not appear automatically as we lock the versions. it is required to bump and use the right version

File: docs/CONTRIBUTING.md [hunk 13]
```diff
@@ -63,3 +63,13 @@ open localhost:6006
     * `wix-style-react/testkit/index.js`
     * `wix-style-react/testkit/enzyme.js`
     * `wix-style-react/testkit/protractor.js`
+
+## Adding new icon
+
+* receive the approval for the new icon from one of the library's UXs [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)
+* go to [wix-ui](https://github.com/wix/wix-ui) repo to [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) package
```
Comment (shlomitc) [inline docs/CONTRIBUTING.md]: make `general` and `system` bold

File: docs/CONTRIBUTING.md [hunk 14]
```diff
@@ -63,3 +63,13 @@ open localhost:6006
...
+* add it to the system icons folder if it's an icon used internally inside a wix-style-react component, or to the general in case it is an icon to be used inside an app
+* add your icon to correct package folder inside [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common)
+* create pull request for [wix-ui](https://github.com/wix/wix-ui) with your new icon
+* after merging your pull request inside [wix-ui](https://github.com/wix/wix-ui), your new icon will automatically appear inside [wix-style-react](https://github.com/wix/wix-style-react) with next wix-style-react release(wix-style-react has automation script, which take all icons from [wix-ui-icons-common](https://github.com/wix/wix-ui/tree/master/packages/wix-ui-icons-common) and proxy them inside [wix-style-react](https://github.com/wix/wix-style-react))
+* your icon will be available in wix-style-react with such syntax `import MyAwesomeIcon from 'wix-style-react/icons/MyAwesomeIcon'` or `import MyAwesomeIcon from 'wix-style-react/icons/system/MyAwesomeIcon'`
```
Comment (shlomitc) [inline docs/CONTRIBUTING.md]: currently should be `new-icons`

File: docs/NEW_ICONS_MIGRATION.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+First of all, some names are changed and some icons are deprecated, so please visit http://electric-process.surge.sh/ and find new icon name, if your icon is deprecated - ask your UX/UI person to provide alternative from available icons.
+
+Now you have two choices:
+1) Do migration by hands based on mapping
+2) run our migration script
+```
+  migrate-to-icons-v2
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: is this how they should actually run it? or use `npx`?

File: docs/NEW_ICONS_MIGRATION.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+```
+such command will migrate all .js files inside src/ folder to new icons,
+if you want to migrate icons in different folder, you can use --path param
+```
+  migrate-to-icons-v2 --path stories/
+```
+
+# Motivation
+
+* current icons do not have standart sizes
```
Comment (shlomitc) [inline docs/NEW_ICONS_MIGRATION.md]: standard

File: package.json [hunk 17]
```diff
@@ -124,7 +130,7 @@
     "slate": "^0.20.3",
     "wix-animations": "^1.0.141",
     "wix-ui-backoffice": "1.0.252",
-    "wix-ui-icons-common": "1.0.39",
+    "wix-ui-icons-common": "^1.0.41",
```
Comment (shlomitc) [inline package.json]: might be a good idea to lock versions anyway

File: src/AutoCompleteComposite/AutoCompleteComposite.js (hunk 4 above)
Comment (shlomitc) [reply src/AutoCompleteComposite/AutoCompleteComposite.js]: ok

**Comment by shlomitc** [comment]
@milkyfruit - what icons size is usually in use in the different applications?
with this icons - everyone will get 24px by default.

**Comment by milkyfruit** [comment]
@shlomitc @Faradey27

24px size is the right size.

All components using general icons have specs for two sets of sizes:
- 24px default and primary size
- 18px size for small actions (right now we don't have 18px icons, so need to scale down default ones)

image
image

What exactly verticals use in their products I don't really know. Probably legacy WSR icons in 'em' size and icons uploaded locally to their projects.

**Comment by Faradey27** [comment]
@shlomitc fixed

File: package.json [hunk 18]
```diff
@@ -124,7 +129,7 @@
     "slate": "^0.20.3",
     "wix-animations": "^1.0.141",
     "wix-ui-backoffice": "1.0.258",
-    "wix-ui-icons-common": "1.0.39",
+    "wix-ui-icons-common": "1.0.41",
```
Comment (shlomitc) [inline package.json]: v2.0.0

File: src/DatePicker/DatePickerDropdown/index.js [hunk 19]
```diff
@@ -63,7 +63,7 @@ export default class DropdownPicker extends React.Component {
             />
 
           <div className={styles.icon}>
-            <ArrowDownThin/>
+            <ChevronDown/>
```
Comment (shlomitc) [inline src/DatePicker/DatePickerDropdown/index.js]: before / after
image / 
image

icon right-left margins

File: src/Dropdown/Dropdown.e2e.js [hunk 20]
```diff
@@ -27,5 +27,5 @@ describe('Dropdown', () => {
         driver.getDropdownItem(3).click();
         expect(driver.getInput().getAttribute('value')).toBe('Option 3');
       });
-  });
+  }, {version: '<Icons/> - use new set of icons'});
```
Comment (shlomitc) [inline src/Dropdown/Dropdown.e2e.js]: same issue here, dropdown suffixes was changed but it shouldn't effect, so let's not change the baseline with a version

File: src/EditableSelector/EditableRow/EditableRow.js [hunk 21]
```diff
@@ -71,7 +71,7 @@ class EditableRow extends WixComponent {
             theme="dark"
             >
             <Button onClick={() => this.onApprove()} height="medium" theme="icon-standard" disabled={this.state.newOption.length === 0} dataHook="edit-row-approve-button">
-              <V size="12px"/>
+              <Check/>
```
Comment (shlomitc) [inline src/EditableSelector/EditableRow/EditableRow.js]: before/after
image
/image

the "+ New Row" was changed and the button V/X icons.

File: src/FieldLabelAttributes/FieldLabelAttributes.js [hunk 22]
```diff
@@ -16,7 +16,7 @@ class FieldLabelAttributes extends WixComponent {
       data-hook="info"
       className={styles.icon}
       >
-      <Info2 size="18px"/>
+      <InfoCircle/>
```
Comment (shlomitc) [inline src/FieldLabelAttributes/FieldLabelAttributes.js]: before/after
image / 
image

padding for all icons was changed

**Comment by Faradey27** [comment]
Was decided to split big pull to several small. So closing this
